packages/xforms-engine/src/lib/reactivity/createBaseItemset.ts

```diff
@@ -0,0 +1,104 @@
+import { UpsertableMap } from '@getodk/common/lib/collections/UpsertableMap.ts';
```

Comment:
Extracted code that `select` and `rank` need from `packages/xforms-engine/src/lib/reactivity/createSelectItems.ts`